src/lib/pmd/PmdLanguageManager.ts

```diff
@@ -0,0 +1,78 @@
+import {Config} from '../util/Config';
+import {Controller} from '../../ioc.config';
+import {PmdEngine} from './PmdEngine';
+
+import {Logger} from '@salesforce/core';
+import {AsyncCreatable} from '@salesforce/kit';
+
+const DEFAULT_LANGUAGES = ['apex', 'javascript'];
+const LANGUAGES_BY_ALIAS: Map<string, string> = new Map([
+	['apex', 'apex'],
+	['java', 'java'],
+	['javascript', 'javascript'],
+	['ecmascript', 'javascript'],
+	['js', 'javascript'],
+	['jsp', 'jsp'],
+	['modelica', 'modelica'],
+	['plsql', 'plsql'],
+	['pl/sql', 'plsql'],
+	['pl-sql', 'plsql'],
+	['scala', 'scala'],
+	['vf', 'visualforce'],
+	['visualforce', 'visualforce'],
+	['xml', 'xml'],
+	['pom', 'xml'],
+	['xsl', 'xml']
+]);
+
+class PmdLanguageManager extends AsyncCreatable {
+	private logger: Logger;
+	private config: Config;
+	private initialized: boolean;
+
+	protected async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		this.logger = await Logger.child('PmdLanguageManager');
+		this.config = await Controller.getConfig();
+		this.initialized = true;
+	}
+
+	private resolveLanguageAliases(aliases: string[]): string[] {
+		const resolvedLangs = [];
+		for (let i = 0; i < aliases.length; i++) {
+			const alias = aliases[i];
+			if (LANGUAGES_BY_ALIAS.has(alias.toLowerCase())) {
+				const lang = LANGUAGES_BY_ALIAS.get(alias.toLowerCase());
+				this.logger.trace(`Resolving language alias ${alias} to ${lang}`);
+				resolvedLangs.push(lang);
+			} else {
+				// TODO: Perhaps this should throw an error?
+				this.logger.trace(`No language found for alias ${alias}`);
+			}
+		}
+		return resolvedLangs;
+	}
+
+	public async getSupportedLanguages(): Promise<string[]> {
+		const pmdConfig = await this.config.getEngineConfig(PmdEngine.NAME);
```

Comment:
Can we keep this logic of reading the correct property from EngineConfig within the Config class? That way all the associated checks can be reused by any piece of code that needs to read this value.

Reply:
@rmohan20 , So would you suggest moving this line into a Config method called something like `getSupportedLanguages()` that accepts an engine name, and keeping the rest the same? Or do you think I should move the entire method into Config?

Reply:
Moving the entire method matches with other config look-ups we have. We've been initializing the default value from Config.ts as well. We should probably standardize the default values in a common place/structure and refer to them from Config class so that it doesn't have to hold the burden of maintaining random default values.

Reply:
Fair enough. I'll do that.

Reply:
Rather, I'll see if I can do that without upending my tests.